docs/dev/GeneratedCode.md

```diff
@@ -0,0 +1,42 @@
+# Testing Generated Code
+
+## Writing Unit Tests For Generated Code
+
+An example of unit testing generated code without running all of Drill is the **priorityQueueOrderingTest()** test in 
+[TopNBatchTest](../../exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/TopN/TopNBatchTest.java). That test tests the 
+[PriorityQueueTemplate](../../exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/TopN/PriorityQueueTemplate.java) class separately from the rest of Drill.
+
+The main accomplished by creating instances of the following classes:
+
+ * [FunctionLookupContext](../../exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionLookupContext.java)
+ * [CodeCompiler](../../exec/java-exec/src/main/java/org/apache/drill/exec/compile/CodeCompiler.java)
+ 
+## Creating A [FunctionLookupContext](../../exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionLookupContext.java)
+
+```
+new FunctionImplementationRegistry(drillConfig)
+```
+
+## Creating A [CodeCompiler](../../exec/java-exec/src/main/java/org/apache/drill/exec/compile/CodeCompiler.java)
+
+ 1. Create a [DrillConfig](../../common/src/main/java/org/apache/drill/common/config/DrillConfig.java) as shown in [MockingComponents.md](MockingComponents.md).
+ 1. Create a [SystemOptionManager](../../exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java) as shown in 
+ [MockingComponents.md](MockingComponents.md).
```

Comment:
I agree that mocking libraries are not a good way to test. I meant this to be a cheat sheet about how to instantiate classes, not a description of how to use a mocking library. I have renamed it to InstantiatingComponents.md to avoid this confusion. I am also updating the document with references to OperatorFixture and ClusterFixture where appropriate.